Core/StorageCache.swift

```diff
@@ -57,24 +63,27 @@ public class StorageCache: StorageCacheUpdating {
             HTTPSUpgrade.shared.loadData()
             return result
             
-        case .httpsBloomFilterSpec:
-            return false
-            
         case .surrogates:
             return fileStore.persist(data as? Data, forConfiguration: configuration)
             
         case .trackerDataSet:
-            if fileStore.persist(data as? Data, forConfiguration: configuration) {
-                if TrackerDataManager.shared.reload(etag: etag) != .downloaded {
-                    Pixel.fire(pixel: .trackerDataReloadFailed)
-                    return false
-                }
-                return true
+            if fileStore.persist(data as? Data, forConfiguration: configuration),
+               TrackerDataManager.shared.reload(etag: etag) != .downloaded {
+                Pixel.fire(pixel: .trackerDataReloadFailed)
+                return false
             }
-            return false
+            return true
```

Comment:
Doesn't that change mean we will be returning true even if persist operation fails?